cirq-superstaq/cirq_superstaq/service.py

```diff
@@ -792,6 +792,85 @@ def process_dfe(self, ids: List[str]) -> float:
         """
         return self._client.process_dfe(ids)
 
+    def submit_aces(
+        self,
+        target: str,
+        qubits: List[int],
+        shots: int,
+        num_circuits: int,
+        mirror_depth: int,
+        extra_depth: int,
+        method: Optional[str] = None,
+        noise: Optional[Union[str, cirq.NoiseModel]] = None,
+        error_prob: Optional[Union[float, Tuple[float, float, float]]] = None,
+        tag: Optional[str] = None,
+        lifespan: Optional[int] = None,
+    ) -> str:
+        """Performs a POST request on the `/aces` endpoint.
+
+        Args:
+            target: The device target to characterize.
+            qubits: A list with the qubit indices to characterize.
+            shots: How many shots to use per circuit submitted.
+            num_circuits: How many random circuits to use in the protocol.
+            mirror_depth: The half-depth of the mirror portion of the random circuits.
+            extra_depth: The depth of the fully random portion of the random circuits.
+            method: Which type of method to execute the circuits with.
+            noise: Noise model to simulate the protocol with. It can be either a string or a
+                `cirq.NoiseModel`.
+
+                Valid strings are "symmetric_depolarize", "phase_flip", "bit_flip" and
+                "asymmetric_depolarize".
+            error_prob: The error probabilities if a string was passed to `noise`.
+
+                For "asymmetric_depolarize", `prob_of_error` will be a three-tuple with the
+                error rates for the X, Y, Z gates in that order. So, a valid argument would be
+                `error_prob = (0.1, 0.1, 0.1)`. notice that these values must add up to less than
+                or equal to 1. For the other channels, `prob_of_error` is one number less than or
+                equal to 1, e.g., `error_prob = 0.1`.
+            tag: Tag for all jobs submitted for this protocol.
+            lifespan: How long to store the jobs submitted for in days (only works with right
+                permissions).
+
+        Returns:
+            A string with the job id for the ACES job created.
+
+        Raises:
+            ValueError: If the target is not valid.
+            ValueError: If the noise model passed is not valid.
+            SuperstaqServerException: if the request fails.
+        """
+        noise_dict: Dict[str, object] = {}
+        if isinstance(noise, str):
+            noise_dict["type"] = noise
+            noise_dict["params"] = error_prob if isinstance(error_prob, tuple) else (error_prob,)
+        elif isinstance(noise, cirq.NoiseModel):
+            noise_dict["cirq_noise_model"] = cirq.to_json(noise)
```

Comment:
So, apparently not **all** `cirq.NoiseModel`s are serializable (see: https://github.com/quantumlib/Cirq/issues/6249). Should I add a `try` statement here to catch when a given noise model is not serializable? Or is the error thrown by Cirq enough?

Error thrown by Cirq:
```
    160 def default(self, o):
    161     """Implement this method in a subclass such that it returns
    162     a serializable object for ``o``, or calls the base implementation
    163     (to raise a ``TypeError``).
   (...)
    177 
    178     """
--> 179     raise TypeError(f'Object of type {o.__class__.__name__} '
    180                     f'is not JSON serializable')

 

TypeError: Object of type InsertionNoiseModel is not JSON serializable

```